### PR TITLE
Archive Page Customization: Enable by default

### LINKED ISSUE #9481: Archive Page Customization: Enable by default

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

This PR enables the new archive page customization feature by default ahead of the planned 1.14.0 release.

## Relevant Technical Choices

Also makes some improvements to the search dropdown to fix some edge cases

## To-do

N/A

## User-facing changes

This settings section is now visible by default:

Screenshot 2021-10-25 at 21 17 34

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

**Default**

1. Visit the settings screen
2. Change archive page setting to "Default"
2. Verify that the `/web-stories/` page lists your latest stories as expected.

**Disabled**

1. Visit the settings screen
2. Change archive page setting to "Disabled"
3. Verify that the `/web-stories/` page now 404s

**Custom**

1. Visit the settings screen
2. Change archive page setting to "Custom"
3. Choose an existing page on your site from the dropdown
3. Verify that the `/web-stories/` page now redirects to that page and displays that page's content

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9481

### CODE REVIEW

**Comment by swissspidy** [comment]
OK I somehow broke the search form test in the `myStories.karma.js` test somehow 🤔 

If anyone has some ideas, that'd be great.

File: packages/design-system/src/components/search/search.js [hunk 1]
```diff
@@ -233,7 +226,10 @@ export const Search = ({
     ]
   );
 
-  useFocusOut(inputRef, handleEndSearch, [handleEndSearch]);
+  // By using inputWrapperRef instead of inputRef we ensure that this isn't triggered
+  // when clicking on the search input's "Clear" button.
+  // It will still trigger when actually clicking on an option within the Popup.
```
Comment (BrittanyIRL) [inline packages/design-system/src/components/search/search.js:231]: 👍

**Review by BrittanyIRL — APPROVED** [review]
Works exactly as described, thank you for the detailed testing directions.

**Review by samwhale — APPROVED** [review]
Nice, works as described!

**Pull request merged**